lib/mp4/transmuxer.js

```diff
@@ -897,7 +962,25 @@ CoalesceStream.prototype.flush = function(flushSource) {
   this.pendingMetadata.length = 0;
 
   // Emit the built segment
+  // We include captions and ID3 tags for backwards compatibility,
+  // ideally we should send only video and audio in the data event
   this.trigger('data', event);
+  // Emit each caption to the outside world
+  // Ideally, this would happen immediately on parsing captions,
+  // but we need to ensure that video data is sent back first
```

Comment:
Do we need to ensure that?

Reply:
We do need the videoTimingInfo to adjust the caption timing correctly.

Reply:
Added a bit more to the comment so we know why